### ISSUE TITLE
Just a quick question

### ISSUE THREAD

**Issue by jenius** [tgriesser/knex#712]
I'm sure there's a good way to do this with knex but I can't quite figure it out. I'm trying to run a query that looks like this:

```sql
SELECT * FROM destinations
INNER JOIN tips
ON tips.destination_id = destinations.id
WHERE tips.private = true
```

So basically just grabbing a relation, but filtering by a single property on the relation. What is the best way to do something like this with knex? Apologies for what is probably a silly question, and really appreciate the time!

**EDIT** I'm sorry, I meant this for bookshelf, let me know if it would be better for me to re-open this in the bookshelf repo. Like with `Destinations.fetchAll({ withRelated: ['tips'] })`, how could I add that second filter layer?

**Comment by tgriesser** [comment]
So are you looking for the tips or the destinations as the result models? 

For what you have there I think this would get it:

```js
Destinations.query(function(qb) {
  qb.innerJoin('tips', 'tips.destination_id', 'destinations.id').where('tips.private', true)
}).fetchAll()
```

I wish GH had a feature of moving tickets between repos haha.

**Comment by jenius** [comment]
Awesome, thanks! This is super helpful.

So I just realized the query I'm working on is a little more complex even than this. I also need to get back a couple other joins in addition to this 'tips' join which includes the private filter. I tried added them in as more `qb.innerJoin` calls in the query block but to no avail. Ended up just manually filtering them before the response is sent from the server.

I'm afraid I'll need to get a little better at SQL and explore the knex/bookshelf code a bit more thoroughly to really get this down, I won't keep asking you to write complex queries for me haha. Thanks for your help though, really appreciate it!
